aips/aip-95.md

```diff
@@ -0,0 +1,195 @@
+     };
+     move_to(&object_signer, collection);
+
+     if (option::is_some(&supply)) {
+         move_to(&object_signer, option::destroy_some(supply))
+     } else {
+         option::destroy_none(supply)
+     };
+
+     if (option::is_some(&royalty)) {
+         royalty::init(&constructor_ref, option::extract(&mut royalty))
+     };
+
+     /* This line is removed to allow for the collection to be transferred.
+        let transfer_ref = object::generate_transfer_ref(&constructor_ref);
+        object::disable_ungated_transfer(&transfer_ref);
+     */
+
+     constructor_ref
+ }
+```
+
+In `token.move`, prior to creating a token, there is a check to verify that the caller is the `creator` of the collection. This will be updated to verify that the caller is the `owner` of the collection object.
+
+```move
+inline fun create_common_with_collection(
+    creator: &signer,
+    constructor_ref: &ConstructorRef,
+    collection: Object<Collection>,
+    description: String,
+    name_prefix: String,
+    // If option::some, numbered token is created - i.e. index is appended to the name.
+    // If option::none, name_prefix is the full name of the token.
+    name_with_index_suffix: Option<String>,
+    royalty: Option<Royalty>,
+    uri: String,
+) {
+    // This line is updated to verify the owner of the collection matches the caller.
+    assert!(object::owner(collection) == signer::address_of(creator), error::unauthenticated(ENOT_OWNER));
+
+    // The rest of the code remains the same...
+}
+```
+
+## Reference Implementations
+
+https://github.com/aptos-labs/aptos-core/pull/14113
+
+## Risks and Drawbacks
+
+**Collection owner not equal to collection creator**
+
+One major potential risk is backwards compatibility. Currently all tokens from a collection are minted via the `creator` of the collection. If there are collections whose `creator` is not the same as the `owner`, this introduces backwards incompatibility and malicious intent.
+
+After collecting relevant data, the result is that there are no collections with a different `creator` and `owner`. This means currently all collections have the same `creator` and `owner`. Therefore introducing this change into the framework would not break backwards compatibility.
+
+- The data was conducted by querying all collections, excluding collections owned by the dead wallet: `0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff`.
+
+**TransferRef not stored for the collection**
+
+As collections are initially created with `ungated_transfer` as false, this means collections can only be transferred with a `TransferRef`.
+If this was not created during construction of the collection, then this collection cannot be transferred to a new owner, meaning no other address can mint tokens from the collection.
+
+To mitigate this, the creation of the collection will not have `ungated_transfer` disabled. This change is reflected above
+in `collection.move` under `Specification`.
```

Comment:
I'm not sure how I feel about this. I think it makes more sense to let this be configurable during collection generation, but I would prefer this remains disabled due to the security properties resulting from it being otherwise.